shared/github/__init__.py

```diff
@@ -48,6 +54,32 @@ def __init__(self, error_cause: InstallationErrorCause, *args: object) -> None:
         self.error_cause = error_cause
 
 
+def decide_installation_error_cause(
+    response: requests.Response,
+) -> InstallationErrorCause:
+    # int(res.headers.get("X-RateLimit-Remaining", -1)) == 0
+    is_primary_rate_limit = int(response.headers.get("X-RateLimit-Remaining", -1)) == 0
+    is_secondary_rate_limit = response.headers.get("Retry-After") is not None
```

Comment:
I’m a bit confused by this, are those two not always equal?
As in: `Retry-After` is only set when `X-RateLimit-Remaining == 0`.

Also not quite sure how python `requests` handles header normalization in terms of lowercasing? I would assume it just does the right thing and internally lowercases everything?

Reply:
About the "lowecasing everything", I got those bits from [here](https://github.com/codecov/shared/blob/gio/debug-installation-error/shared/torngit/github.py#L865)... and that seems to be working... I should test those things just to be sure though.

I don't see why it should lowercase headers that start with `X-*` tho?

Reply:
I didn't find evidence that `requests` lowercases the headers (testing locally, I mean)